fms2_io/fms_netcdf_domain_io.F90

```diff
@@ -767,6 +768,18 @@ subroutine get_global_io_domain_indices(fileobj, dimname, is, ie)
       call error("input dimension is not associated with the domain.")
     endif
   endif
+
+  if (present(indices)) then 
+    if(allocated(indices)) then
+      deallocate(indices)
+    endif
+    allocate(indices(ie-is+1))
```

Comment:
It seems odd that you are reallocating a variable that's sent in.  This seems like some behind the scenes magic sauce that could cause danger.

Reply:
@uramirez8707 If you change `indices` to `out`, then then if it's already allocated, I think you should call a fatal instead of just plowing over whatever was already sent in.  The caller should be handing you an unallocated array.